scripts/zones/Yuhtunga_Jungle/npcs/Blue_Rafflesia.lua

```diff
@@ -0,0 +1,132 @@
+        if (evenmoreTravelsStatus == QUEST_ACCEPTED and questprogress == 1 and player:getVar("SecondBlueRafflesiaCS") == 0) then
+            player:startEvent(0x0016);
+        elseif (evenmoreTravelsStatus == QUEST_COMPLETED and scentReady == true and player:getVar("BathedInScent") == 0 and player:getVar("SecondBlueRafflesiaCS") == 0) then
+            player:startEvent(0x0016);
+        else
+            player:messageSpecial(FLOWER_BLOOMING);
+        end
+    elseif (npcId == 17281588) then
+        if (evenmoreTravelsStatus == QUEST_ACCEPTED and questprogress == 1 and player:getVar("ThirdBlueRafflesiaCS") == 0) then
+            player:startEvent(0x0017);
+        elseif (evenmoreTravelsStatus == QUEST_COMPLETED and scentReady == true and player:getVar("BathedInScent") == 0 and player:getVar("ThirdBlueRafflesiaCS") == 0) then
+            player:startEvent(0x0017);
+        else
+            player:messageSpecial(FLOWER_BLOOMING);
+        end
+    end
+end;
+
+-----------------------------------
+-- onEventUpdate
+-----------------------------------
+
+function onEventUpdate(player,csid,option)
+--printf("CSID: %u",csid);
+--printf("OPTION: %u",option);
+end;
+
+-----------------------------------
+-- onEventFinish
+-----------------------------------
+
+function onEventFinish(player,csid,option)
+--printf("CSID: %u",csid);
+--printf("OPTION: %u",option);
+    local evenmoreTravelsStatus = player:getQuestStatus(OUTLANDS, EVEN_MORE_GULLIBLES_TRAVELS)
+
+-- Set BathedInScent to 1 if they touched all 3 Rafflesia for EVEN_MORE_GULLIBLES_TRAVELS which opens the quest The Opo-Opo and I
+    if (csid == 0x0015 and option == 1) then
+        if (player:getVar("SecondBlueRafflesiaCS") == 1 and player:getVar("ThirdBlueRafflesiaCS") == 1) then -- This is 3rd Rafflessia clicked, progressing.
+            player:setVar("SecondBlueRafflesiaCS", 0);
+            player:setVar("ThirdBlueRafflesiaCS", 0);
+            player:setVar("BathedInScent", 1);
+            player:setVar("RafflesiaScentDay",VanadielDayOfTheYear());
+            player:setVar("RafflesiaScentYear",VanadielYear());
+            player:addItem(1144);        -- Rafflesia Nectar
+            player:messageSpecial(ITEM_OBTAINED,1144);
+            player:messageSpecial(FEEL_DIZZY); -- You feel slightly dizzy. You must have breathed in too much of the pollen.
+            if (evenmoreTravelsStatus == QUEST_ACCEPTED) then
+                player:setVar("EVEN_MORE_GULLIBLES_PROGRESS", 2);
+            end
+        else
+            player:setVar("FirstBlueRafflesiaCS", 1);
+            player:addItem(1144);        -- Rafflesia Nectar
+            player:messageSpecial(ITEM_OBTAINED,1144);
+        end
+    elseif (csid == 0x0016 and option == 1) then
+        if (player:getVar("FirstBlueRafflesiaCS") == 1 and player:getVar("ThirdBlueRafflesiaCS") == 1) then
+            player:setVar("FirstBlueRafflesiaCS", 0);
+            player:setVar("ThirdBlueRafflesiaCS", 0);

```

Comment:
instead of using 3 separate vars, use a bitmask

e.g. in the first one, set it to
```
    local rafflesiaCS = player:getVar("RafflesiaCS");
    bit.bor(rafflesiaCS, 0x01); -- set the bit for first cs
```

in the second one, do the same but set it to 0x02, and in the third do the same but set it to 0x04 (start at 0x01 for flagged, then multiply each subsequent value by 2 for each additional mask e.g. 0x01, 0x02, 0x04, 0x08)

once you've set each flag, you can check by adding the bits you want to check and comparing using `bit.band(mask, bits to check)` e.g. 
```
if bit.band(rafflesiaCS, 0x01) then -- first cs only
    -- do shit
elseif bit.band(rafflesiaCS, 0x05) then -- we want to check both the first and third have been triggered, so add the bits for first (0x01) + third (0x04) = 0x05
    -- do shit
```